dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/RequestParamsValidator.java

```diff
@@ -336,63 +336,62 @@ else if (filters.length == 4) {
    * @throws BadRequestException if a wrong combination of org unit and org unit mode supplied
    */
   public static OrganisationUnitSelectionMode validateOrgUnitMode(
-      Set<UID> orgUnits, OrganisationUnitSelectionMode orgUnitMode) throws BadRequestException {
+      Set<UID> orgUnits,
+      OrganisationUnitSelectionMode orgUnitMode,
+      Set<UID> trackedEntities,
+      boolean validateTrackedEntities)
+      throws BadRequestException {
 
     if (orgUnitMode == null) {
       return orgUnits.isEmpty() ? ACCESSIBLE : SELECTED;
     }
 
-    if (!orgUnits.isEmpty()
-        && (orgUnitMode == ACCESSIBLE || orgUnitMode == CAPTURE || orgUnitMode == ALL)) {
+    if (orgUnitModeDoesNotRequireOrgUnit(orgUnitMode) && !orgUnits.isEmpty()) {
       throw new BadRequestException(
           String.format(
               "orgUnitMode %s cannot be used with orgUnits. Please remove the orgUnit parameter and try again.",
               orgUnitMode));
     }
 
-    if ((orgUnitMode == CHILDREN || orgUnitMode == SELECTED || orgUnitMode == DESCENDANTS)
-        && orgUnits.isEmpty()) {
-      throw new BadRequestException(
-          String.format(
-              "At least one org unit is required for orgUnitMode: %s. Please add an orgUnit or use a different orgUnitMode.",
-              orgUnitMode));
+    if (validateTrackedEntities) {
+      validateOrgUnitOrTrackedEntityIsPresent(orgUnitMode, orgUnits, trackedEntities);
+    } else {
+      validateOrgUnitIsPresent(orgUnitMode, orgUnits);
     }
 
     return orgUnitMode;
   }
 
-  /**
-   * Validates that the org unit is not present if the ou mode is ACCESSIBLE or CAPTURE. If it is,
-   * an exception will be thrown. If the org unit mode is not defined, SELECTED will be used by
-   * default if an org unit is present. Otherwise, ACCESSIBLE will be the default.
-   *
-   * @param orgUnit the org unit to validate
-   * @return a valid org unit mode
-   * @throws BadRequestException if a wrong combination of org unit and org unit mode supplied
-   */
-  public static OrganisationUnitSelectionMode validateOrgUnitMode(
-      UID orgUnit, OrganisationUnitSelectionMode orgUnitMode) throws BadRequestException {
-
-    if (orgUnitMode == null) {
-      orgUnitMode = orgUnit != null ? SELECTED : ACCESSIBLE;
-    }
-
-    if ((orgUnitMode == ACCESSIBLE || orgUnitMode == CAPTURE) && orgUnit != null) {
+  private static void validateOrgUnitOrTrackedEntityIsPresent(
```

Comment:
Is this a change in the behavior?
Why if there are TE Uids in the request, we can have empty orgUnits even for orgUnitModes that require orgUnits?